### PR TITLE
Add custom components

### PR DESCRIPTION
**Pull request by AkshatGKMit**
**Components List**
- Icon and Icon button component
- Text input component
- Custom button component
- Loader component
- Radio button component
- Text button component

And refactor prettier configuration

### CODE REVIEW

File: src/components/TextButton/TextButton.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  const handlePressIn = () => setIsPressIn(true);
+
+  const handlePressOut = () => setIsPressIn(false);
+
+  return (
+    <View style={styles.outsideView}>
+      <TouchableWithoutFeedback
+        onPress={isDisabled ? undefined : onPress}
+        onPressIn={isDisabled ? undefined : handlePressIn}
```
Comment (02priyanshi) [inline src/components/TextButton/TextButton.tsx:17]: why define isDisabled for onPress seperately when you have already used disabled property

File: src/components/TextButton/TextButton.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        onPressOut={isDisabled ? undefined : handlePressOut}
+        disabled={isDisabled}
+        {...addOns}
+      >
+        <View>
+          <Text
+            style={[
+              styles.text,
+              customStyles,
+              {
```
Comment (02priyanshi) [inline src/components/TextButton/TextButton.tsx:27]: remove inline styles

File: src/components/TextField/styles.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    fontWeight: "700",
+    fontSize: 16,
+  },
+  textfield: {
+    borderBottomWidth: 2,
+    flexDirection: "row",
+    alignItems: "center",
+    gap: 7,
+    height: 42,
+    borderColor: "darkgrey",
```
Comment (02priyanshi) [inline src/components/TextField/styles.ts:19]: import from color constant

**Review by 02priyanshi — CHANGES_REQUESTED** [review]

File: src/components/GestureDetector/styles.ts [hunk 4]
```diff
@@ -2,27 +2,27 @@ import { StyleSheet } from "react-native";
...
+    shadowOffset: {
+      width: 0,
+      height: 2,
+    },
+    shadowOpacity: 0.25,
+    shadowRadius: 3.84,
+    elevation: 5,
+  },
+  text: {
+    color: "white",
```
Comment (02priyanshi) [inline src/components/GestureDetector/styles.ts]: color constant

File: src/components/GestureDetector/GestureDetector.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+}: GestureDetectorProps) => {
+  return (
+    <TouchableOpacity
+      onPress={isLoading ? undefined : onPress}
+      disabled={isDisabled || isLoading}
+      activeOpacity={0.75}
+      {...addOns}
+    >
+      <View style={[styles.button, customStyles]}>
+        {isLoading ? <Loader size={20} /> : isString(children) ? <Text style={styles.text}>{children}</Text> : children}
```
Comment (02priyanshi) [inline src/components/GestureDetector/GestureDetector.tsx:22]: use prettier

File: src/components/TextButton/TextButton.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+const TextButton = ({ text, addOns, isDisabled = false, styles: customStyles, onPress }: TextButtonProps) => {
+  return (
+    <TouchableWithoutFeedback
+      onPress={onPress}
+      disabled={isDisabled}
+      {...addOns}
+    >
+      <View>
+        <Text style={[styles.text, customStyles, isDisabled && styles.isDisabled]}>{text}</Text>
```
Comment (02priyanshi) [inline src/components/TextButton/TextButton.tsx]: use prettier

File: src/components/TextField/styles.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    borderRadius: 6,
+    paddingHorizontal: 7,
+    paddingVertical: 2,
+  },
+  textInput: {
+    flex: 1,
+    height: "100%",
+    width: "100%",
+    fontSize: 16,
+    color: "black",
```
Comment (02priyanshi) [inline src/components/TextField/styles.ts]: use color constant

**Review by 02priyanshi — CHANGES_REQUESTED** [review]

File: src/components/GestureDetector/GestureDetector.tsx (hunk 5 above)
Comment (AkshatGKMit) [reply src/components/GestureDetector/GestureDetector.tsx:22]: My prettier print width is set to 125
so this line is under 125 character

File: src/components/TextButton/TextButton.tsx (hunk 6 above)
Comment (AkshatGKMit) [reply src/components/TextButton/TextButton.tsx]: My prettier print width is set to 125
so this line is under 125 character


will apply ternary in this
`isDisabled && styles.isDisabled`

**Review by 02priyanshi — APPROVED** [review]

**Pull request merged**